src/java-interop/java-interop-gc-bridge-mono.c

```diff
@@ -155,11 +160,36 @@ ji_realpath (const char *path)
 	if (path == NULL)
 		return NULL;
 
+#ifndef WINDOWS
 	char   *rp = realpath (path, NULL);
 	if (rp == NULL) {
 		return strdup (path);
 	}
 	return rp;
+#else
+#define BUFSIZE 4096
+
+	wchar_t *wpath = utf8_to_utf16 (path);
+	char *fullpath;
+	wchar_t buffer [BUFSIZE];
+	DWORD retval = GetFullPathNameW (wpath, BUFSIZE, buffer, NULL);
+	assert (retval > 0);
+
+	if (retval >= BUFSIZE) {
+		wchar_t *allocated_buffer = malloc (sizeof (wchar_t)*retval);
```

Comment:
This should use `xmalloc()` (which may need migrating from xamarin-android) or you need to NULL-check `allocated_buffer`.